docs/edgeql/group.rst

```diff
@@ -0,0 +1,195 @@
+.. _ref_eql_group:
+
+Group
+=====
+
+EdgeQL supports a top-level ``group`` statement. These are used to partition
+sets into subsets based on some parameters. These subsets then can be
+additionally aggregated to provide some analytics.
+
+The most basic format is just using the bare :eql:stmt:`group` to group a set
+of objects by some property:
+
+.. code-block:: edgeql-repl
+
+    db> with M := (select Movie filter .release_year > 2015)
+    ... group M{title} by .release_year;
+    {
+      {
+        key: {release_year: 2016},
+        grouping: {'release_year'},
+        elements: {
+          default::Movie {title: 'Captain America: Civil War'},
+          default::Movie {title: 'Doctor Strange'},
+        },
+      },
+      {
+        key: {release_year: 2017},
+        grouping: {'release_year'},
+        elements: {
+          default::Movie {title: 'Spider-Man: Homecoming'},
+          default::Movie {title: 'Thor: Ragnarok'},
+        },
+      },
+      {
+        key: {release_year: 2018},
+        grouping: {'release_year'},
+        elements: {default::Movie {title: 'Ant-Man and the Wasp'}},
+      },
+      {
+        key: {release_year: 2019},
+        grouping: {'release_year'},
+        elements: {default::Movie {title: 'Spider-Man: No Way Home'}},
+      },
+      {
+        key: {release_year: 2021},
+        grouping: {'release_year'},
+        elements: {default::Movie {title: 'Black Widow'}},
+      },
+    }
+
+In the above example referring to the property in the ``by`` clause **must**
+be done by using ``.release_year`` short path. The property name then shows up
+in ``grouping`` and ``key`` to indicate the defining characteristics of the
```

Comment:
I think we should emphasize that the result is a set of free shapes since that's a very "EdgeDB-ish" aspect to the design. We should break down what each of the keys (`key`, `grouping`, and `elements`) in a bulleted list.

Reply:
Added more explanation.